package vfsprotocol;

message PingMessage
{
	required string sender_id = 1;
}

message ReadRequest
{
	required string object_id = 1;
	required uint32 object_type = 2;
	required uint64 size = 3;
	required uint64 offset = 4;
}

message WriteRequest
{
	required string object_id = 1;
	required uint32 object_type = 2;
	required uint64 size = 3;
	required uint64 offset = 4;
}

message WriteResponse
{
	required uint64 size = 1;
	optional bool dtl_in_sync = 2 [default = false];
}

message SyncRequest
{
	required string object_id = 1;
	required uint32 object_type = 2;
}

message SyncResponse
{
	optional bool dtl_in_sync = 1 [default = false];
}

message GetSizeRequest
{
	required string object_id = 1;
	required uint32 object_type = 2;
}

message GetSizeResponse
{
	required uint64 size = 1;
}

message GetClusterMultiplierRequest
{
	required string object_id = 1;
	required uint32 object_type = 2;
}

message GetClusterMultiplierResponse
{
	required uint32 size = 1;
}

message GetCloneNamespaceMapRequest
{
	required string object_id = 1;
	required uint32 object_type = 2;
}

message GetCloneNamespaceMapResponse
{
    message MapEntry
    {
        required uint32 clone_id = 1;
        required string ns = 2;
    }
    repeated MapEntry map_entry = 3;
}

message GetPageRequest
{
    required string object_id = 1;
    required uint32 object_type = 2;
    required uint64 cluster_address = 3;
}

message GetPageResponse
{
    repeated uint64 cluster_location = 1;
}

message ResizeRequest
{
	required string object_id = 1;
	required uint32 object_type = 2;
	required uint64 size = 3;
}

message DeleteRequest
{
	required string object_id = 1;
	required uint32 object_type = 2;
}

message TransferRequest
{
	required string object_id = 1;
	required uint32 object_type = 2;
	required string target_node_id = 3;
	optional uint64 sync_timeout_ms = 4;
}

message OpenRequest
{
	required string object_id = 1;
	required uint32 object_type = 2;
}

// Local Variables: **
// mode: protobuf **
// End: **
